Importing libraies and reading data csv file into a Pandas dataframe

In [1]:
import pandas as pd
import numpy as np

In [2]:
# The code was removed by Watson Studio for sharing.

In [3]:
df_tor = pd.read_csv(body)
df_tor.head()

Postcode           Borough     Neighbourhood
0      M1A      Not assigned      Not assigned
1      M2A      Not assigned      Not assigned
2      M3A        North York         Parkwoods
3      M4A        North York  Victoria Village
4      M5A  Downtown Toronto      Harbourfront

Rename columns

In [4]:
df_tor.rename(columns={'Postcode':'PostalCode', 'Neighbourhood':'Neighborhood'}, inplace=True)
df_tor.head()

PostalCode           Borough      Neighborhood
0        M1A      Not assigned      Not assigned
1        M2A      Not assigned      Not assigned
2        M3A        North York         Parkwoods
3        M4A        North York  Victoria Village
4        M5A  Downtown Toronto      Harbourfront

Remove all rows that have a Borough Not Assigned

In [5]:
df_tor['Borough'].replace('Not assigned',np.nan, inplace=True)
df_tor.dropna(subset=['Borough'], axis=0, inplace=True)
df_tor.head()

PostalCode           Borough      Neighborhood
2        M3A        North York         Parkwoods
3        M4A        North York  Victoria Village
4        M5A  Downtown Toronto      Harbourfront
5        M6A        North York  Lawrence Heights
6        M6A        North York    Lawrence Manor

Combine rows with the same Postal Code <br/>
Replace Not Assigned Neighborood values with their corresponding Borough value <br/>
Randomize dataframe rows

In [6]:
df_tor=df_tor.groupby(['PostalCode','Borough'], as_index =False).agg({'Neighborhood': ', '.join})
df_tor['Neighborhood'].replace('Not assigned',df_tor['Borough'], inplace=True)
df_tor = df_tor.sample(frac=1).reset_index(drop=True)
df_tor.head()

PostalCode       Borough                                       Neighborhood
0        M3J    North York                    Northwood Park, York University
1        M9C     Etobicoke  Bloordale Gardens, Eringate, Markland Wood, Ol...
2        M4L  East Toronto                     The Beaches West, India Bazaar
3        M9L    North York                                      Humber Summit
4        M9A     Etobicoke                                   Islington Avenue

In [7]:
df_tor.shape

(103, 3)

The datframe has 103 rows or records and 3 columns or features

In [8]:
body = client_d6b108fb8f9a44b6b683b7059b7338a1.get_object(Bucket='myfirstproject-donotdelete-pr-qxcxkzyfq9az74',Key='Geospatial_Coordinates.csv')['Body']
# add missing __iter__ method, so pandas accepts body as file-like object
if not hasattr(body, "__iter__"): body.__iter__ = types.MethodType( __iter__, body )

# If you are reading an Excel file into a pandas DataFrame, replace `read_csv` by `read_excel` in the next statement.
df_coords = pd.read_csv(body)
df_coords.head()

Postal Code   Latitude  Longitude
0         M1B  43.806686 -79.194353
1         M1C  43.784535 -79.160497
2         M1E  43.763573 -79.188711
3         M1G  43.770992 -79.216917
4         M1H  43.773136 -79.239476

Join the coordinates dataframe with the original dataframe on the matching Postal Code of both dataframes

In [11]:
df_tor=df_tor.join(df_coords.set_index('Postal Code'), on='PostalCode')
df_tor.head(12)

PostalCode           Borough  \
0         M3J        North York   
1         M9C         Etobicoke   
2         M4L      East Toronto   
3         M9L        North York   
4         M9A         Etobicoke   
5         M1M       Scarborough   
6         M5M        North York   
7         M3H        North York   
8         M9B         Etobicoke   
9         M3M        North York   
10        M1B       Scarborough   
11        M5J  Downtown Toronto   

                                         Neighborhood   Latitude  Longitude  
0                     Northwood Park, York University  43.767980 -79.487262  
1   Bloordale Gardens, Eringate, Markland Wood, Ol...  43.643515 -79.577201  
2                      The Beaches West, India Bazaar  43.668999 -79.315572  
3                                       Humber Summit  43.756303 -79.565963  
4                                    Islington Avenue  43.667856 -79.532242  
5     Cliffcrest, Cliffside, Scarborough Village West  43.716316 -79.239476  
6                   Bedford Park, Lawrence Manor East  43.733283 -79.419750  
7     Bathurst Manor, Downsview North, Wilson Heights  43.754328 -79.442259  
8   Cloverdale, Islington, Martin Grove, Princess ...  43.650943 -79.554724  
9                                   Downsview Central  43.728496 -79.495697  
10                                     Rouge, Malvern  43.806686 -79.194353  
11  Harbourfront East, Toronto Islands, Union Station  43.640816 -79.381752